python_modules/libraries/dagster-powerbi/dagster_powerbi/resource.py

```diff
@@ -437,7 +437,7 @@ def fetch_state(self) -> PowerBIWorkspaceData:
             )
 
     def defs_from_state(self, state: PowerBIWorkspaceData) -> Definitions:
-        translator = self.translator_cls(context=state)
+        translator = self.translator_cls().copy_with_context(context=state)
```

Comment:
State-backed defs and spec loader to accept a translator instance in a subsequent PR.